src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -139,12 +149,37 @@ export function DiscussionThread(props: DiscussionThreadProps) {
       <FormControl type="text" placeholder="Write a reply..." />
 
       {hasNextPage ? (
-        <Button variant="outline-primary" onClick={() => fetchNextPage()}>
+        <Button
+          variant="outline-primary"
+          onClick={() => {
+            fetchNextPage()
+          }}
+        >
           Show more results
         </Button>
       ) : (
         <></>
       )}
+      <Modal
+        size="lg"
+        show={threadModal}
+        onHide={() => setThreadModal(false)}
+        animation={false}
+      >
+        <Modal.Header>
+          <Modal.Title>New Thread</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <MarkdownEditor
+            onSave={onSave}
+            forumId={forumId}
+            onCancel={() => setThreadModal(false)}
+            editTitle={threadData?.title}
+            editText={threadBody}
+          />
```

Comment:
This should use a ForumThreadEditor component. See comments on MarkdownEditor.tsx for more details